tracer/src/Datadog.Trace/Ci/ImpactedTestsModule.cs

```diff
@@ -2,188 +2,226 @@
+                // Milestone 1.5 : Retrieve diff files and lines from Git Diff CLI but with the base commit from the backend (always try the maximum accuracy)
+                try
+                {
+                    modifiedFiles = GitCommandHelper.GetGitDiffFilesAndLines(workspacePath, baseCommitSha!, currentCommitSha);
+                }
+                catch (Exception ex)
                 {
-                    if (string.IsNullOrEmpty(testFile.Path)) { continue; }
+                    Log.Debug(ex, "Git command failed.");
+                }
+            }
 
-                    var modifiedFile = modifiedFiles.FirstOrDefault(x => testFile.Path!.EndsWith(x.Path!));
-                    if (modifiedFile is not null)
+            // If we still don't have any modified files, we use the ones from the backend
+            if (modifiedFiles.Length == 0)
+            {
+                Log.Debug("ImpactedTestsModule: Retrieving diff files from backend...");
+                if (impactedTestsDetectionResponse is { Files: { Length: > 0 } files })
+                {
+                    modifiedFiles = new FileCoverageInfo[files.Length];
+                    for (var x = 0; x < files.Length; x++)
                     {
-                        Log.Debug("DiffFile found {File} ...", modifiedFile.Path);
-
-                        if (testFile.ExecutedBitmap is null || modifiedFile.ExecutedBitmap is null)
+                        if (string.IsNullOrEmpty(files[x]))
                         {
-                            Log.Debug("  No line info");
-                            modified = true;
-                            break;
+                            continue;
                         }
 
-                        var testFileBitmap = new FileBitmap(testFile.ExecutedBitmap);
-                        var modifiedFileBitmap = new FileBitmap(modifiedFile.ExecutedBitmap);
-
-                        if (testFileBitmap.IntersectsWith(ref modifiedFileBitmap))
-                        {
-                            Log.Debug("  Intersecting lines. Marking test {TestName} as modified.", test.Name);
-                            modified = true;
-                            break;
-                        }
+                        modifiedFiles[x] = new FileCoverageInfo(files[x]);
                     }
                 }
-
-                if (modified)
-                {
-                    tags.IsModified = "true";
-                    TelemetryFactory.Metrics.RecordCountCIVisibilityImpactedTestsIsModified();
-                }
-            }
-            else
-            {
-                Log.Debug("Impacted Tests Detection is DISABLED for {TestName}", test.Name);
             }
         }
-        catch (Exception err)
+
+        if (modifiedFiles.Length == 0)
         {
-            Log.Error(err, "Error analyzing Impacted Tests for {TestName}", test.Name);
+            Log.Warning("ImpactedTestsModule: No modified files found.");
```

Comment:
If no changes were detected (could be a normal situation) wouldn't it be better to issue an Info instead of a Warning?

Reply:
I guess it could happen in an empty commit. Becase this is extracting diff from a base sha. I'll change it to Info

Reply:
done in [5f9af43](https://github.com/DataDog/dd-trace-dotnet/pull/6836/commits/5f9af43d7a736be8458910f6b3fd8979e52fdca9)